core/vm/memory_table.go

```diff
@@ -48,6 +48,10 @@ func memoryMStore(stack *Stack) (uint64, bool) {
 	return calcMemSize64WithUint(stack.Back(0), 32)
 }
 
+func memoryMemoryCopy(stack *Stack) (uint64, bool) {
+	return calcMemSize64(stack.Back(0), stack.Back(2))
```

Comment:
This doesn's seem right. It's supposed to calculate memory size after the operation's expansion, and this will return `dst+length`, but `src` could be above `dst`, so it should be
```suggestion
	return calcMemSize64(max(stack.Back(0), stack.Back(1)), stack.Back(2))
```

Reply:
thanks for spotting that